actor/todo/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+use std::collections::HashMap;
+
+use anyhow::{anyhow, Result};
+use log::{info, trace, warn};
+use serde::{Deserialize, Serialize};
+use wasmbus_rpc::actor::prelude::*;
+use wasmcloud_interface_httpserver::{HttpRequest, HttpResponse, HttpServer, HttpServerReceiver};
+use wasmcloud_interface_keyvalue::{
+    IncrementRequest, KeyValue, KeyValueSender, SetAddRequest, SetDelRequest, SetRequest,
+};
+#[derive(Debug, Default, Actor, HealthResponder)]
+#[services(Actor, HttpServer)]
+struct TodoActor {}
+
+// FIXME: contribute this upstream
+struct HttpResponseShim {}
+impl HttpResponseShim {
+    /// Creates a response with a given status code and serializes the given payload as JSON
+    pub fn json<T>(payload: T, status_code: u16) -> HttpResponse
+    where
+        T: Serialize,
+    {
+        HttpResponse {
+            body: serde_json::to_string(&payload).unwrap().into_bytes(),
+            header: HashMap::new(),
+            status_code,
+        }
+    }
+    /// Handy shortcut for creating a 404/Not Found response
+    pub fn not_found() -> HttpResponse {
+        HttpResponse {
+            status_code: 404,
+            ..Default::default()
+        }
+    }
+    /// Useful shortcut for creating a 200/OK response
+    pub fn ok() -> HttpResponse {
+        HttpResponse {
+            status_code: 200,
+            ..Default::default()
+        }
+    }
+
+    /// Useful shortcut for creating a 500/Internal Server Error response
+    pub fn internal_server_error(msg: &str) -> HttpResponse {
+        HttpResponse {
+            status_code: 500,
+            body: msg.to_string().as_bytes().into(),
+            ..Default::default()
+        }
+    }
+    // and also:
+    // /// Shortcut for creating a 400/Bad Request response
+    // pub fn bad_request() -> HttpResponse {
+    //     HttpResponse {
+    //         status_code: 400,
+    //         ..Default::default()
+    //     }
+    // }
+}
+
+/// Implementation of HttpServer trait methods
+#[async_trait]
+impl HttpServer for TodoActor {
```

Comment:
I think I'm going to move this to the bottom of the file, where it was in the original todo-backend example. I much prefer this order of defining things, because I'm a C programmer at heart.